src/prefect/task_worker.py

```diff
@@ -413,7 +413,6 @@ def status():
     return status_app
 
 
-@sync_compatible
```

Comment:
this `serve` is now async-only which feels weird given the other methods are sync - what's the reasoning?

Reply:
great call - my fault, I meant to do this here as well

updated in [7db3991](https://github.com/PrefectHQ/prefect/pull/14855/commits/7db39911394acc62046a8e0a1311cef5740ad572)